test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -314,6 +351,78 @@ def test_fsdp_tp_integration(self, tensor_parallel_size, cpu_offload):
         tp_fsdp_out = tp_fsdp_model(inp)
         self.assertEqual(fsdp_out, tp_fsdp_out)
 
+    @skip_if_lt_x_gpu(4)
+    def test_fsdp_tp_gradient_layout(self):
+        """
+        Tests TP + FSDP extension with consistent gradient layout
+        """
+        mesh_1d = init_device_mesh("cuda", (self.world_size,))
+
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.mlp = MLPModule("cuda")
+                self.mlp_norm = RMSNormPython(10)
+
+            def forward(self, x):
+                return self.mlp_norm(self.mlp(x))
+
+        model = TestModel().cuda(self.rank)
+
+        # Shard with TP and test gradient
+        tp_model = parallelize_module(
+            model,
+            mesh_1d,
+            {
+                "mlp.net1": ColwiseParallel(),
+                "mlp.net2": RowwiseParallel(output_layouts=Shard(0)),
+            },
+        )
+        distribute_rmsnorm(tp_model.mlp_norm, mesh_1d)
```

Comment:
IIUC, this means the RMSNorm runs as "sequence parallel"?

Reply:
Yep, this is sequence parallel layernorm/rmsnorm, the backward would produce partial dtensor, and the parameter would be replicated dtensor